plugins/cloud-gce/src/main/java/org/elasticsearch/plugin/cloud/gce/CloudGcePlugin.java

```diff
@@ -56,7 +59,7 @@ public String description() {
     @Override
     public Collection<Module> nodeModules() {
         List<Module> modules = new ArrayList<>();
-        if (settings.getAsBoolean("cloud.enabled", true)) {
+        if (settings.getAsBoolean("cloud.enabled", true) && GceModule.isDiscoveryReady(settings, logger)) {
```

Comment:
Not really related to this change but we can remove now the check for cloud.enabled